Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+                                stepActivity.Expected = StripHTML(step.Expected);
+                                stepActivity.ActivityName = tc.TestName + ">" + step.StepName;
+                            }
+                            else//not in ActivitiesGroup so get instance from repo
+                            {
+                                stepActivity = (Activity)((Activity)repoStepActivity).CreateInstance(true);
+                                toAddStepActivity = true;
+                            }
+                        }
+                        else//Step not exist in Ginger repository so create new one
+                        {
+                            stepActivity = new Activity();
+                            stepActivity.ActivityName = step.Description;
+                            stepActivity.ExternalID = step.StepID;
+                            stepActivity.Description = step.Description;
+                            stepActivity.Expected = step.Expected;
+
+                            toAddStepActivity = true;
+                        }
+
+                        if (toAddStepActivity)
+                        {
+                            //not in group- need to add it
+                            busFlow.AddActivity(stepActivity, tcActivsGroup);                            
+                        }
+
+                        //pull TC-Step parameters and add them to the Activity level
+                        List<string> stepParamsList = new List<string>();
+                        GetStepParameters(StripHTML(step.Description), ref stepParamsList);
+                        GetStepParameters(StripHTML(step.Expected), ref stepParamsList);
+                        foreach (string param in stepParamsList)
+                        {
+                            //get the param value
+                            string paramSelectedValue=string.Empty;
+                            bool? isflowControlParam=null;
+                            QtestTestParameter tcParameter = tc.Parameters.Where(x => x.Name.ToUpper() == param.ToUpper()).FirstOrDefault();
+
+                            //get the param value
+                            if (tcParameter != null && tcParameter.Value != null && tcParameter.Value != string.Empty)
+                                paramSelectedValue = tcParameter.Value;
+                            else
+                            {
+                                isflowControlParam = null;//empty value
+                                paramSelectedValue = "<Empty>";
+                            }
+
+                            //check if parameter is part of a link
+                            string linkedVariable = null;
+                            if (paramSelectedValue.StartsWith("#$#"))
+                            {
+                                string[] valueParts = paramSelectedValue.Split(new string[] {"#$#"}, StringSplitOptions.None);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)